File: src/arduinobot_py_examples/arduinobot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        super().__init__('simple_publisher')
        self.publisher_ = self.create_publisher(String, 'example_topix', 10)
        self.frequency_ = 1.0  # seconds
        self.get_logger().info(f'Publishing at {self.frequency_} Hz')
        self.timer_ = self.create_timer(self.frequency_, self.publishMessage)
        self.counter_ = 0

    def publishMessage(self):
        msg = String()
        msg.data = f'Hello from ROS2 - {self.counter_}'
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publishing: "{msg.data}"')
        self.counter_ += 1

def main(args=None):
    rclpy.init()
    simple_publisher = SimplePublisher()
    rclpy.spin(simple_publisher)
    simple_publisher.detroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/arduinobot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <chrono>

using namespace std::chrono_literals;

class SimplePublisher : public rclcpp::Node
{
    public:
        SimplePublisher() : Node("simple_publisher")
        {
            publisher_ = create_publisher<std_msgs::msg::String>("example_topix", 10);
            timer_ = create_wall_timer(1s, std::bind(&SimplePublisher::publishMessage, this));
            
            RCLCPP_INFO(get_logger(), "Publishing at 1 Hz");
        }

        void publishMessage()
        {
            auto message = std_msgs::msg::String();
            message.data = "Hello from ROS2 - " + std::to_string(counter_++);
            publisher_->publish(message);
            RCLCPP_INFO(get_logger(), "Publishing: '%s'", message.data.c_str());
        }

    private:
        unsigned int counter_ = 0;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc, char* argv[])
{
    rclcpp::init(argc,argv);
    auto simplePublisher = std::make_shared<SimplePublisher>();
    rclcpp::spin(simplePublisher);
    rclcpp::shutdown();
    return 0;
}